src/app/shared/components/tree/tree.component.html

```diff
@@ -0,0 +1,32 @@
+<mat-tree [dataSource]="dataSource" [treeControl]="treeControl"
+  class="govuk-!-margin-bottom-6">
+  <!-- nodes without children -->
+  <mat-tree-node
+    class="tree-node"
+    [ngClass]="{ 'is-active': node.isActive }"
+    *matTreeNodeDef="let node"
+    matTreeNodeToggle
+    matTreeNodePadding
+  >
+    <button class="govuk-!-margin-left-6" (click)="setActiveNode(node)" mat-icon-button>
+      {{ node.nodeName }}
+    </button>
+  </mat-tree-node>
+
+  <!-- nodes that have children -->
+  <mat-tree-node class="tree-node"
+    [ngClass]="{ 'is-active': node.isActive }"
+    *matTreeNodeDef="let node; when: hasChildren" matTreeNodePadding>
+    <button
+      (click)="setActiveNode(node)"
```

Comment:
To double check with Dragos but we are very likely to need to add id attributes
like test-tree-button-{i} for automation testers please